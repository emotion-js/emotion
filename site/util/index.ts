export * from './style-constants'
export * from './dracula-prism'
