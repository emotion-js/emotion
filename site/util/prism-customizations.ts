import Prism from 'prismjs'
import 'prismjs/components/prism-jsx'
import 'prismjs/components/prism-typescript'
import 'prismjs/components/prism-tsx'

// Prevent Prism from automatically highlighting code, which is unnecessary and
// causes hydration warnings.
;(Prism as { manual?: boolean }).manual = true

// This is only necessary for the live editors, for some reason it "just works"
// with remark-prism. But, we make sure this is imported on every page so that
// the Prism configuration doesn't depend on whether there is a live editor
// present.
Prism.languages.insertBefore('jsx', 'template-string', {
  'styled-template-string': {
    pattern:
      /(styled(\.\w+|\([^)]*\))(\.\w+(\([^)]*\))*)*|css|injectGlobal|keyframes|css={)`(?:\$\{[^}]+\}|\\\\|\\?[^\\])*?`/,
    lookbehind: true,
    greedy: true,
    inside: {
      interpolation: {
        pattern: /\$\{[^}]+\}/,
        inside: {
          'interpolation-punctuation': {
            pattern: /^\$\{|\}$/,
            alias: 'punctuation'
          },
          rest: Prism.languages.jsx
        }
      },
      string: {
        pattern: /[^$;]+/,
        inside: Prism.languages.css,
        alias: 'language-css'
      }
    }
  }
})
