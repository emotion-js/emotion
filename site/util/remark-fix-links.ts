import { visit } from 'unist-util-visit'

export function remarkFixLinks() {
  return (markdownAST: any) => {
    visit(markdownAST, 'link', (node: { url: string }) => {
      node.url = node.url.replace(/^https?:\/\/emotion.sh/i, '')
      node.url = node.url.replace(
        /^https?:\/\/github.com\/emotion-js\/emotion\/tree\/main/i,
        ''
      )

      if (!node.url.startsWith('http')) {
        node.url = node.url
          .replace(/\.mdx?(#.*)?$/, (_, hash) => {
            return hash || ''
          })
          .replace(/^\/packages\//, '/docs/@emotion/')
      }
    })
  }
}
