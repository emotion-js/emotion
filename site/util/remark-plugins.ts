import { visit } from 'unist-util-visit'

export function remarkFixLinks() {
  return (markdownAST: any) => {
    visit(markdownAST, 'link', (node: { url: string }) => {
      node.url = node.url.replace(/^https?:\/\/emotion.sh/, '')

      if (!node.url.startsWith('//') && !node.url.startsWith('http')) {
        node.url = node.url
          .replace(/\.mdx?(#.*)?$/, (_, hash) => {
            return hash || ''
          })
          .replace(/^\/packages\//, '/docs/@emotion/')
      }
    })
  }
}

export function remarkResponsiveTables() {
  return (markdownAST: any) => {
    visit(markdownAST, 'mdxJsxFlowElement', (node: any) => {
      if (node.name === 'table') {
        node.name = 'ResponsiveTable'
      }
    })
  }
}
