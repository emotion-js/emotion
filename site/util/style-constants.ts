export const colors = {
  body: '#212529',
  pink: '#D26AC2',
  hightlight: '#C43BAD',
  pinkBorder: '#eac3e4',
  grayBorder: '#d2d2d2',
  pinkBg: '#FBF8FD',
  hotPink: '#c865b9',
  danger: '#dc3545',
  dangerBorder: '#f5c2c7',

  // See https://github.com/twbs/bootstrap/blob/main/scss/_variables.scss if
  // other grays are needed
  gray100: '#f8f9fa',
  gray500: '#adb5bd'
}

const breakpoints = { sm: 576, md: 768, lg: 992, xl: 1200, xxl: 1400 }

export const mediaQueries = {
  smUp: `@media (min-width: ${breakpoints.sm}px)`,
  mdUp: `@media (min-width: ${breakpoints.md}px)`,
  lgUp: `@media (min-width: ${breakpoints.lg}px)`,
  xlUp: `@media (min-width: ${breakpoints.xl}px)`,
  xxlUp: `@media (min-width: ${breakpoints.xxl}px)`
}

export const styleConstants = {
  borderRadius: '0.25rem',

  fontSizeSm: '0.875rem',
  fontSizeLg: '1.25rem'
}
