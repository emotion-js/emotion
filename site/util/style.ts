export const colors = {
  body: '#212529',
  pink: '#D26AC2',
  //   blue: '#46C9E5',
  //   yellow: '#FFCA2A',
  //   dark: '#1D2029',
  //   reallyLightPink: '#f9e9fb',
  hightlight: '#C43BAD',
  border: '#eac3e4'
  // border: '#C865B9'
  //   bg: '#FAF9FA',
  //   color: '#1B1D1D',
  //   parentBg: '#FBF8FD'
}

const breakpoints = { sm: 576, md: 768, lg: 992, xl: 1200, xxl: 1400 }

export const mediaQueries = {
  smUp: `@media (min-width: ${breakpoints.sm}px)`,
  mdUp: `@media (min-width: ${breakpoints.md}px)`,
  lgUp: `@media (min-width: ${breakpoints.lg}px)`,
  xlUp: `@media (min-width: ${breakpoints.xl}px)`,
  xxlUp: `@media (min-width: ${breakpoints.xxl}px)`
}
