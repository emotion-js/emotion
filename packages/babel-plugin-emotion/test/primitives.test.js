// @flow

import { createInlineTests } from './util'

const cases = {
  'does not change to a call expression when beginning with a upper case letter': {
    code: `
    styled.View\`color: hotpink;\`
    styled.View({})
    `
  },
  'other name with @emotion/primitives': {
    code: `
    import someOtherName from '@emotion/primitives'
    someOtherName.View\`color: hotpink;\`
    someOtherName.View({})
    `
  },
  'does not change call expressions': {
    code: `
    styled('SomeFakeComponent')\`color: hotpink;\`
    styled('SomeFakeComponent')({})
    `
  }
}

createInlineTests('styled inline', cases)
