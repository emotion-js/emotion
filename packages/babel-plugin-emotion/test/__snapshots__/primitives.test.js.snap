// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled inline babel 6 does not change call expressions 1`] = `
"
/*#__PURE__*/styled('SomeFakeComponent', {
    target: 'e8nbyc50'
})('color:hotpink;');
/*#__PURE__*/styled('SomeFakeComponent', {
    target: 'e8nbyc51'
})({});"
`;

exports[`styled inline babel 6 does not change to a call expression when beginning with a upper case letter 1`] = `
"
/*#__PURE__*/styled.View(\\"color:hotpink;\\");
/*#__PURE__*/styled.View({});"
`;

exports[`styled inline babel 6 other name with @emotion/primitives 1`] = `
"
import someOtherName from '@emotion/primitives';
/*#__PURE__*/someOtherName.View('color:hotpink;');
/*#__PURE__*/someOtherName.View({});"
`;

exports[`styled inline babel 7 does not change call expressions 1`] = `
"/*#__PURE__*/
styled('SomeFakeComponent', {
  target: \\"e1u6ll0g0\\"
})(\\"color:hotpink;\\");

/*#__PURE__*/
styled('SomeFakeComponent', {
  target: \\"e1u6ll0g1\\"
})({});"
`;

exports[`styled inline babel 7 does not change to a call expression when beginning with a upper case letter 1`] = `
"/*#__PURE__*/
styled.View(\\"color:hotpink;\\");

/*#__PURE__*/
styled.View({});"
`;

exports[`styled inline babel 7 other name with @emotion/primitives 1`] = `
"import someOtherName from '@emotion/primitives';

/*#__PURE__*/
someOtherName.View(\\"color:hotpink;\\");

/*#__PURE__*/
someOtherName.View({});"
`;
